slick-codegen/src/main/scala/slick/codegen/AbstractSourceCodeGenerator.scala

```diff
@@ -93,19 +129,24 @@ def $name($args): $name = {
 
     trait PlainSqlMapperDef extends super.PlainSqlMapperDef{
       def code = {
-        val positional = compoundValue(columnsPositional.map(c => (if(c.fakeNullable || c.model.nullable)s"<<?[${c.rawType}]"else s"<<[${c.rawType}]")))
+        def argList(values: Seq[String]) = values.mkString("(" ,", " , ")")
+        val positional = columnsPositional.map(c => (if(c.fakeNullable || c.model.nullable)s"<<?[${c.rawType}]"else s"<<[${c.rawType}]"))
         val dependencies = columns.map(_.exposedType).distinct.zipWithIndex.map{ case (t,i) => s"""e$i: GR[$t]"""}.mkString(", ")
-        val rearranged = compoundValue(desiredColumnOrder.map(i => if(hlistEnabled) s"r($i)" else tuple(i)))
-        def result(args: String) = if(mappingEnabled) s"$factory($args)" else args
+        val rearranged   =if( desiredColumnOrder.size <= maxTupleSize)
+                            desiredColumnOrder.map(i => tuple(i))
+                          else
+                            desiredColumnOrder.zipWithIndex.map{ case(index,i) => s"${tuple(index/maxTupleSize)}.${tuple(index%maxTupleSize)}" }
```

Comment:
this also looks like it would not work for tables > 440  columns, because for column 441 you would get _23._1